assets/app/scripts/directives/nav.js

```diff
@@ -53,11 +53,16 @@ angular.module('openshiftConsole')
 
           sortedProjects = $filter('orderByDisplayName')(projects);
 
-          options = _.map(sortedProjects, function(item) {
+          options = _.map(sortedProjects, function(item, index) {
```

Comment:
Agreed. I'll use the approach of a filter. What about updating the displayName filter ? I think it's preferable over the creation of a new one.